setup.rst

```diff
@@ -385,17 +385,29 @@ for the header and library files to your ``configure`` command.  For example,
 
 with **Homebrew**::
 
-    $ brew install pkg-config openssl xz gdbm
+    $ brew install pkg-config openssl xz gdbm tcl-tk
 
-and ``configure`` Python versions >= 3.7::
+For Python 3.11 and newer::
 
-    ./configure --with-pydebug --with-openssl=$(brew --prefix openssl)
+    $ PKG_CONFIG_PATH="$(brew --prefix tcl-tk)/lib/pkgconfig" \
+      ./configure --with-pydebug --with-openssl=$(brew --prefix openssl)
 
-or ``configure`` Python versions < 3.7::
+For Python versions 3.10 through 3.7::
 
+    $ export PKG_CONFIG_PATH="$(brew --prefix tcl-tk)/lib/pkgconfig"
+    $ ./configure --with-pydebug \
+                  --with-openssl=$(brew --prefix openssl) \
+                  --with-tcltk-libs="$(pkg-config --libs tcl tk)" \
+                  --with-tcltk-includes="$(pkg-config --cflags tcl tk)"
```

Comment:
Tested on 3.10 through 3.7 ✅

Reply:
Can you confirm you don't get this warning?

```
./configure --with-pydebug --with-openssl=$(brew --prefix openssl@1.1) --with-tcltk-libs="$(pkg-config --libs tcl tk)" --with-tcltk-includes="$(pkg-config --cflags tcl tk)" | grep 'unrecognized options'
configure: WARNING: unrecognized options: --with-tcltk-libs, --with-tcltk-includes
```

Reply:
@bxsx I am not getting the warning anymore. I was on `main` instead of 3.10 when getting the error as suggested by @erlend-aasland. 

This is the full output I am getting when `make -s -j2` after running your `configure` command:
<details>

> /Library/Developer/CommandLineTools/usr/bin/ranlib: file: libpython3.10d.a(dynamic_annotations.o) has no symbols
> /Library/Developer/CommandLineTools/usr/bin/ranlib: file: libpython3.10d.a(pymath.o) has no symbols
>  CC='gcc' LDSHARED='gcc -bundle -undefined dynamic_lookup    ' OPT='-g -O0 -Wall' 	_TCLTK_INCLUDES='-I/usr/local/Cellar/tcl-tk/8.6.12_1/include' _TCLTK_LIBS='-L/usr/local/Cellar/tcl-tk/8.6.12_1/lib -ltk8.6 -ltkstub8.6 -ltcl8.6 -ltclstub8.6' 	./python.exe -E ./setup.py -q build
> ld: warning: directory not found for option '-L/usr/lib/termcap'
> /Users/izumiberat/Documents/GitHub/cpython/Modules/_sqlite/module.c:141:10: warning: 
>       'sqlite3_enable_shared_cache' is deprecated: first deprecated in macOS
>       10.7 - Not supported [-Wdeprecated-declarations]
>     rc = sqlite3_enable_shared_cache(do_enable);
>          ^
> /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sqlite3.h:5950:16: note: 
>       'sqlite3_enable_shared_cache' has been explicitly marked deprecated here
> SQLITE_API int sqlite3_enable_shared_cache(int);
>                ^
> 1 warning generated.
> In file included from /Users/izumiberat/Documents/GitHub/cpython/Modules/_tkinter.c:46:
> In file included from /usr/local/Cellar/tcl-tk/8.6.12_1/include/tk.h:99:
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:131:21: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         int (*free_private)();  /* called to free private storage */
>                            ^
>                             void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:334:33: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         struct _XImage *(*create_image)();
>                                        ^
>                                         void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:453:23: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         XID (*resource_alloc)(); /* allocator function */
>                              ^
>                               void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:471:20: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         int (*synchandler)();   /* Synchronization handler */
>                           ^
>                            void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:496:24: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         Bool (*event_vec[128])();  /* vector for wire to event */
>                               ^
>                                void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:497:25: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         Status (*wire_vec[128])(); /* vector for event to wire */
>                                ^
>                                 void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:509:20: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         Bool (**error_vec)();      /* vector for wire to error */
>                           ^
>                            void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:522:25: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         int (*savedsynchandler)(); /* user synchandler when Xlib usurps */
>                                ^
>                                 void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:1053:24: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
> typedef void (*XIMProc)();
>                        ^
>                         void
> 9 warnings generated.
> In file included from /Users/izumiberat/Documents/GitHub/cpython/Modules/tkappinit.c:17:
> In file included from /usr/local/Cellar/tcl-tk/8.6.12_1/include/tk.h:99:
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:131:21: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         int (*free_private)();  /* called to free private storage */
>                            ^
>                             void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:334:33: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         struct _XImage *(*create_image)();
>                                        ^
>                                         void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:453:23: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         XID (*resource_alloc)(); /* allocator function */
>                              ^
>                               void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:471:20: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         int (*synchandler)();   /* Synchronization handler */
>                           ^
>                            void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:496:24: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         Bool (*event_vec[128])();  /* vector for wire to event */
>                               ^
>                                void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:497:25: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         Status (*wire_vec[128])(); /* vector for event to wire */
>                                ^
>                                 void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:509:20: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         Bool (**error_vec)();      /* vector for wire to error */
>                           ^
>                            void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:522:25: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
>         int (*savedsynchandler)(); /* user synchandler when Xlib usurps */
>                                ^
>                                 void
> /usr/local/Cellar/tcl-tk/8.6.12_1/include/X11/Xlib.h:1053:24: warning: this
>       function declaration is not a prototype [-Wstrict-prototypes]
> typedef void (*XIMProc)();
>                        ^
>                         void
> 9 warnings generated.
> 
> Python build finished successfully!
> The necessary bits to build these optional modules were not found:
> ossaudiodev           spwd                                     
> To find the necessary bits, look in setup.py in detect_modules() for the module's name.
> 
> 
> The following modules found by detect_modules() in setup.py, have been
> built by the Makefile instead, as configured by the Setup files:
> _abc                  pwd                   time               
> 
> renaming build/scripts-3.10/pydoc3 to build/scripts-3.10/pydoc3.10
> renaming build/scripts-3.10/idle3 to build/scripts-3.10/idle3.10
> renaming build/scripts-3.10/2to3 to build/scripts-3.10/2to3-3.10

</details>

Reply:
> @bxsx I am not getting the warning anymore. I was on `main` instead of 3.10 when getting the error as suggested by @erlend-aasland.

Oh, you are both right! Sorry, that was my fault. I confirm it works as expected!